frida/src/script.rs

```diff
@@ -5,46 +5,167 @@
  */
 
 use frida_sys::{FridaScriptOptions, _FridaScript, g_bytes_new, g_bytes_unref};
+use serde::Deserialize;
+use serde_json::Value;
+use std::cell::Cell;
 use std::marker::PhantomData;
+use std::sync::mpsc::{channel, Receiver, Sender};
 use std::{
     ffi::{c_char, c_void, CStr, CString},
     ptr::null_mut,
 };
 
 use crate::{Error, Result};
 
+/// Represents a Frida message
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type")]
+#[serde(rename_all = "lowercase")]
+pub enum Message {
+    /// Message of type "send"
+    Send(MessageSend),
+
+    /// Message of type "log"
+    Log(MessageLog),
+
+    /// Message of type "error"
+    Error(MessageError),
+
+    /// Any other type of message.
+    Other(Value),
+}
+
+/// Send Message.
+#[derive(Deserialize, Debug)]
+pub struct MessageSend {
+    /// Payload of a Send Message.
+    pub payload: SendPayload,
+}
+
+/// Log Message.
+#[derive(Deserialize, Debug)]
+pub struct MessageLog {
+    /// Log Level.
+    pub level: MessageLogLevel,
+    /// Payload of a Message Log.
+    pub payload: String,
+}
+
+/// Error message.
+/// This message is sent when a JavaScript runtime error occurs, such as a misspelled word.
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct MessageError {
+    /// Error description.
+    pub description: String,
+
+    /// Stack trace string.
+    pub stack: String,
+
+    /// Script file name that failed.
+    pub file_name: String,
+
+    /// Line number with the error.
+    pub line_number: usize,
+
+    /// Column number with the error.
+    pub column_number: usize,
+}
+
+/// Represents a Message Log Level Types.
+/// Used by `MessageLog._level`
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "lowercase")]
+pub enum MessageLogLevel {
+    /// Indicates an informal message.
+    Info,
+    /// Represents a debugging message.
+    Debug,
+    /// Signifies a warning message.
+    Warning,
+    /// Represents an error message.
+    Error,
+}
+
+/// Represents a Message Log Level Types.
+/// Used by `MessageLog._level`
+#[derive(Deserialize, Debug)]
+pub struct SendPayload {
+    /// Send message type
+    pub r#type: String,
+    /// Send message ID
+    pub id: usize,
+    /// Send message result.
+    pub result: String,
+    /// Send message returns.
+    pub returns: Value,
+}
+
 unsafe extern "C" fn call_on_message<I: ScriptHandler>(
     _script_ptr: *mut _FridaScript,
     message: *const i8,
     _data: &frida_sys::_GBytes,
     user_data: *mut c_void,
 ) {
-    let handler: &mut I = &mut *(user_data as *mut I);
+    let c_msg = CStr::from_ptr(message as *const c_char)
+        .to_str()
+        .unwrap_or_default();
+
+    let formated_msg: Message = serde_json::from_str(c_msg).unwrap_or_else(|err| {
```

Comment:
`formatted_msg`

Reply:
I wanted to show the different types of messages. That's why in the readme I added the output to expect. I think it is easier to show this kind of things for newcomers, but I can remove it if you want to.